File: src/backend/Gerenciamento.Informacoes.ESocial/Gerenciamento.Informacoes.ESocial.Dominio/Entidades/Trabalhador.cs
using Gerenciamento.Informacoes.ESocial.Dominio.Enums;
using Gerenciamento.Informacoes.ESocial.Dominio.ObjetosValor;

namespace Gerenciamento.Informacoes.ESocial.Dominio.Entidades;

public class Trabalhador
{
    public int TrabalhadorId { get; set; }
    public TipoVinculo Tipo { get; set; }
    public string? Nome { get; set; }
    public Sexo Sexo { get; set; }
    public RacaCor RacaCor { get; set; }
    public EstadoCivil EstadoCivil { get; set; }
    public GrauInstrucao GrauInstrucao { get; set; }
    public bool IsPrimeiroEmprego { get; set; }
    public string? CodigoNomeTravTrans { get; set; }
    public DateTime? DataNascimento { get; set; }
    public string? MunicipioNascimento { get; set; }
    public string? UfNascimento { get; set; }
    public string? PaisNascimento { get; set; }
    public string? Nacionalidade { get; set; }
    public string? NomeMae { get; set; }
    public string? NomePai { get; set; }
    public Documento? DocumentosPessoais { get; set; }
    public Endereco? EnderecoResidencial { get; set; }
    public Deficiencia? DadosDeficiencia { get; set; }
    public bool RecebeBeneficioPrevidencia { get; set; }
    public Contato? Contato { get; set; }
    public bool IsVerificado { get; set; }
    public DateTime DataCadastro { get; set; } = DateTime.Now;
    public DateTime? DataUltimaAtualizacao { get; set; }
    public  ICollection<Dependente>? Dependentes { get; set; }
    public ICollection<Estagiario>? Estagiarios { get; set; }
    public ICollection<Cedido>? Cedidos { get; set; }
}

File: src/frontend/gerenciamento-esocial-app/models/Trabalhador.ts
import { Contato } from "./Contato";
import { Dependente } from "./Dependente";
import { Documento } from "./Documento";
import { Endereco } from "./Endereco";
import { Deficiencia } from "./Deficiencia";

// ====== Tipos ======
export type TipoVinculo = 1 | 2 | 3; // CLT=1, Estagiário=2, Outro=3
export type Sexo = "Masculino" | "Feminino" | "";
export type RacaCor = "Branco" | "Negro" | "Pardo" | "Indígena" | "Amarelo" | "";
export type EstadoCivil = "Solteiro" | "Casado" | "Divorciado" | "Viúvo" | "";
export type GrauInstrucao = "Fundamental" | "Medio" | "Superior" | "Pos" | "Mestrado" | "Doutorado" | "";
export type StatusCadastro = "Criado" | "Pendente" | "Concluido";

export interface TrabalhadorForm {
  nome: string;
  tipo: TipoVinculo;
  sexo: Sexo;
  racaCor: RacaCor;
  estadoCivil: EstadoCivil;
  grauInstrucao: GrauInstrucao;
  isPrimeiroEmprego: boolean;
  codigoNomeTravTrans: string;
  dataNascimento: string;
  municipioNascimento: string;
  ufNascimento: string;
  paisNascimento: string;
  nacionalidade: string;
  nomeMae: string;
  nomePai: string;
  documentosPessoais?: Documento;
  enderecoResidencial?: Endereco;
  dadosDeficiencia?: Deficiencia;
  recebeBeneficioPrevidencia: boolean;
  contato?: Contato;
  statusCadastro: StatusCadastro;
  possuiDependente: boolean;
  dependentes: Dependente[];
}